libs/remote_storage/src/local_fs.rs

```diff
@@ -28,29 +23,32 @@ const LOCAL_FS_TEMP_FILE_SUFFIX: &str = "___temp";
 
 #[derive(Debug, Clone)]
 pub struct LocalFs {
-    storage_root: PathBuf,
+    storage_root: Utf8PathBuf,
 }
 
 impl LocalFs {
     /// Attempts to create local FS storage, along with its root directory.
     /// Storage root will be created (if does not exist) and transformed into an absolute path (if passed as relative).
-    pub fn new(mut storage_root: PathBuf) -> anyhow::Result<Self> {
+    pub fn new(mut storage_root: Utf8PathBuf) -> anyhow::Result<Self> {
         if !storage_root.exists() {
             std::fs::create_dir_all(&storage_root).with_context(|| {
                 format!("Failed to create all directories in the given root path {storage_root:?}")
             })?;
         }
         if !storage_root.is_absolute() {
-            storage_root = storage_root.canonicalize().with_context(|| {
-                format!("Failed to represent path {storage_root:?} as an absolute path")
-            })?;
+            storage_root = storage_root
+                .canonicalize()
+                .with_context(|| {
+                    format!("Failed to represent path {storage_root:?} as an absolute path")
+                })
+                .map(|pb| Utf8PathBuf::from_path_buf(pb).unwrap())?;
```

Comment:
Perhaps https://docs.rs/camino/latest/camino/struct.Utf8Path.html#method.canonicalize_utf8 should be used here?